### PR TITLE
Add GPU support

### PR DESCRIPTION
**Pull request by odow**
This PR is a rebase of #187 onto the `jump-dev` version of SCS.jl so we can run the JuliaGPU tests.

I followed the instructions here: https://github.com/JuliaGPU/gitlab-ci, and I think I have things set up correctly. I added a check for `GITLAB_CI` so that we only run the GPU tests on JuliaGPU, and default to the main tests on Travis.

Closes #187

### CODE REVIEW

**Comment by odow** [comment]
@kalmarek ideas? https://gitlab.com/JuliaGPU/SCS-jl/-/jobs/791684624#L169

File: test/test_gpu.jl [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # This file requires a GPU in order to run. It gets tested as part of the
 # JuliaGPU CI on Gitlab. Contact @odow for more details.
 
+# CUDA_jll is not in our Project.toml, so we need to install it on GITLAB_CI.
+import Pkg
+Pkg.add("CUDA_jll")
```
Comment (kalmarek) [inline test/test_gpu.jl:6]:
```suggestion
pkg"add CUDA_jll@9.0"
```
I actually don't known how to use `PackageSpec` with `Pkg.add` to achieve this ;)
`SCS_GPU` was built against this version:

https://github.com/JuliaPackaging/Yggdrasil/blob/01037ee342dd40ea57c3ad20e69ad8bc81d24bbe/S/SCS_GPU/build_tarballs.jl#L38

This was an arbitrary choice, I picked the lowest available to support possibly widest range of GPUs

**Review by kalmarek — COMMENTED** [review]
hopefully this will do the trick

**Comment by odow** [comment]
Now we have the fun error:

> ERROR: LoadError: LoadError: ArgumentError: Package Pkg not found in current path:
>  Run `import Pkg; Pkg.add("Pkg")` to install the Pkg package.

https://gitlab.com/JuliaGPU/SCS-jl/-/jobs/793855432#L276

**Comment by kalmarek** [comment]
It's marvelously recurrent :D tbh I don't know what to do about this, but I asked on slack ;)

File: README.md [hunk 2]
```diff
@@ -77,7 +78,32 @@ optimize!(problem)
...
+julia> using SCS
+
+julia> SCS.available_solvers
+2-element Array{DataType,1}:
+ SCS.DirectSolver
+ SCS.IndirectSolver
+```
+
+```julia
+julia> using CUDA_jll # optional, to enable SCS.GpuIndirectSolver
```
Comment (odow) [inline README.md:96]: TODO: add instructions on how to add the correct version of CUDA_jll

**Comment by odow** [comment]
Success! I didn't setup codecov yet. It looks like I need to give Gitlab some secret. But we're passing tests at least, and the linear system solves are on a sparse-indirect GPU:
https://gitlab.com/JuliaGPU/SCS-jl/-/jobs/794254537#L3441

**Comment by kalmarek** [comment]
that is great! I didn't notice that we had separate `[test]` category.

As for tests: maybe we could turn `for T in solvers` loop in `test/MOI_wrapper.jl` into a function of `T` and just call it here with `SCS.GpuIndirectSolver`?

**Comment by kalmarek** [comment]
go or no go?

Is it breaking enough to justify major version bump? technically speaking no officially supported parts were altered, but we require new packages. I'm fine eitherway

**Comment by odow** [comment]
Should be fine as a patch release. As you say, this exclusively adds new features and doesn't break any existing code.

**Comment by kalmarek** [comment]
when CI is green, we're good to go and merge this ;)

**Pull request merged**